internal/provider/baremetal/installConfig.go

```diff
@@ -42,14 +43,34 @@ func (p baremetalProvider) AddPlatformToInstallConfig(
 			return err
 		}
 
+		if len(cluster.MachineNetworks) == 0 {
+			err = errors.Errorf("Failed to find machine networks for baremetal cluster %s", cluster.ID)
+			p.Log.Error(err)
+			return err
+		}
+
 		for _, iface := range inventory.Interfaces {
-			if iface.MacAddress != "" {
+			// We are looking for the NIC that matches the first Machine Network configured
+			// for the cluster. This is to ensure that BootMACAddress belongs to the NIC that
+			// is really used and not to any fake interface even if this interface has IPs.
+			match := false
+			ipAddressses := append(iface.IPV4Addresses, iface.IPV6Addresses...)
+			for _, ip := range ipAddressses {
+				overlap, _ := network.NetworksOverlap(ip, string(cluster.MachineNetworks[0].Cidr))
+				if overlap {
+					match = true
+				}
```

Comment:
Tiny suggestion to reuse the `match` variable and to break out of this loop early if a match is found
```suggestion
				if match, _ = network.NetworksOverlap(ip, string(cluster.MachineNetworks[0].Cidr)); match {
					// maybe even set this here?
					// hosts[yamlHostIdx].BootMACAddress = iface.MacAddress
					break
				}
```

Reply:
That's a very good suggestion, thanks! Looks like I was trying to over-engineer and completely missed something as trivial as this